CMakeLists.txt

```diff
@@ -152,6 +152,26 @@ if(Cabana_ENABLE_COVERAGE_BUILD)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
 endif()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  option(WITH_ASAN "Build with address sanitizer" OFF)
+endif()
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT APPLE)
+  option(WITH_MSAN "Build with memory sanitizer (experimental; requires a memory-sanitized Python interpreter)" OFF)
+endif()
+
+if(WITH_ASAN AND WITH_MSAN)
+  message( FATAL_ERROR "Address sanitizer and memory sanitizer cannot be enabled simultaneously")
+endif()
+if(WITH_ASAN)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+endif()
+
+if(WITH_MSAN)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=memory -fno-omit-frame-pointer")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=memory")
+endif()
```

Comment:
Why an option when all you need is CXX flags?

Reply:
Only clang support `-fsanitize=memory` and some obscure compilers might not have `-fsanitize=address`.